src/components/MapView/MapView.tsx

```diff
@@ -42,31 +56,20 @@ const MapView = forwardRef<MapViewHandle, ComponentProps>(
                     return;
                 }
 
+                if (!shouldPanMapToCurrentPosition()) {
+                    setCurrentPositionToInitialState();
+                    return;
+                }
+
                 hasAskedForLocationPermission.current = true;
```

Comment:
@s77rt I don't think we should modify this here because imagine a case where the `MapView` has a waypoint so it doesn't request location then the waypoint is deleted and thus we need to pan to current location. At that point, we would need to request the user location, which is only possible if `hasAskedForLocationPermission` is `false`. And technically we are not exactly asking for permission, we are just setting it to the initial state. This shouldn't prevent further attempts for asking the permission.

Reply:
@s77rt Please do let me know if the current approach leads to any bug or unexpected result.

Reply:
> imagine a case where the MapView has a waypoint so it doesn't request location then the waypoint is deleted and thus we need to pan to current location

Is this something on `main` already?

> Please do let me know if the current approach leads to any bug or unexpected result

No specific bug yet but I found out that setCurrentPositionToInitialState is being called more than once

Reply:
> Is this something on main already?
I'm not sure, which is why I'm hesitant to making this change as it may break in a case where we are doing so...

> No specific bug yet but I found out that setCurrentPositionToInitialState is being called more than once
Yes, `setCurrentPositionToInitialState` was just the error callback of the `getCurrentPosition` so, I figured we need to mirror the behaviour of declination of the location position when we don't require it. And not setting `hasAskedForLocationPermission` to `true` will make sure that we are able to request the permission when it is actually required.

Reply:
I think we need to set the position to the user's location only at the start, if the user already has waypoints we shouldn't ask him for location permission even if he deleted those waypoints later. Is this something against `main`?

Reply:
@s77rt What I understand from the above comment is that you are proposing we should call `setCurrentPositionToInitialState` once in the beginning of `useEffect` regardless of the `offline` or `hasAskedForLocationPermission`, right?
If yes, I think that can cause problems with `main` because the `currentPosition` would get defined before the `getCurrentPosition` function executes. Which means that the following line will be changed:
https://github.com/Expensify/App/blob/669ac330b4eeaad2238419bdb501811d3e031e13/src/components/MapView/MapView.tsx#L145
Thus it will prevent the `pendingView` from rendering which should render when we ask the user for their location permission.

Reply:
It is important that `currentPosition` stays undefined before we ask for permission so that pending view is displayed.

Reply:
No, I'm only proposing to set hasAskedForLocationPermission to true if we are going to change the current position (as seen in the code diff)

Reply:
Okay, I'll make the change if you insist and should we rename it to `shouldAskForLocationPermission`? (start as true and set to false in the `useEffect`) or should we leave the current name?

Reply:
bump @s77rt I'm available to make the change now itself..

Reply:
Rename to `shouldInitializeCurrentPosition`